src/scenario/outdoors.cpp

```diff
@@ -199,3 +199,8 @@ bool cOutdoors::cWandering::isNull(){
 void cOutdoors::reattach(cScenario& scen) {
 	scenario = &scen;
 }
+
+bool cOutdoors::is_city_on_map(spec_loc_t loc)
```

Comment:
I don't understand, why are you checking `x` and not `y`? Shouldn't it be more like...

```
return loc.spec > 0 && is_on_map(loc);
```

Reply:
Because it is a direct refactor from ```sector.city_locs[k].spec >= 0 && sector.city_locs[k].x < 48``` on line 1261 on the commit preceding this branch
I'm not sure this is the right name for this method now that I read this, see comments on your review of is_tile_on_edge

Reply:
https://github.com/calref/cboe/blob/master/src/game/boe.town.cpp#L1261-L1264

If you look at those lines, I think you'll find the x<0 check is actually superfluous when combined with the minmax tests.

So I guess instead of having a function you could just directly test whatever.spec>=0 in the source function (ie, inline).

Reply:
I mean I suppose you could add an `is_valid()` function to `spec_loc_t` that returns `spec >= 0`, but... that seems a bit much to me.

Still, I'm particularly not opposed to it, if you prefer that.